lib/bap_powerpc/bap_powerpc.mli

```diff
@@ -0,0 +1,675 @@
+
+(**
+
+    {2 Intro}
+
+    This module is the only one that needed to write lifter functions.
+    The main idea is to make a life of lifter writers as easy as
+    possible. This implies that only few lines of code should be
+    enough to describe any instruction.
+
+    We introduce RTL - the language we expect to be very expressive,
+    with lots of details hidden under the hood, so a user should not
+    care about minor details.
+
+    So proposed usage is just to open at the very beginning of your
+    module:
+
+    {[
+      open Bap_powerpc.Std
+    ]}
+
+    {2 RTL}
+
+    The central part of this module is RTL. It contains expressions,
+    operations over expressions and statements.
+    Basically, any line that ended with ';' is a (compound) statement,
+    and any part of it is either a statement, expression(s), or operation over
+    expression(s).
+
+    {3 Bitwidth and Signedness}
+
+    Any expression in RTL has a notion of signedness and bitwidth.
+
+    If an operation is applied to expressions with different signs,
+    any unsigned expression is implicitly casted to signed one.
+
+    If an operation is applied to expressions with different bitwidths,
+    an expression with the smaller bitwidth is implicitly casted
+    (extended) to the bitwidth of another expression.
+
+    {3 Expressions}
+
+    There are only few ways to construct an expression:
+     - from instruction operand
+     - from constant
+     - from string
+     - by defining temporary variable
+
+    To construct an expression that denotes a register and treat
+    its content as an unsigned value, one should write:
+
+    {v
+     let ra = unsigned cpu.reg op.(0)
+              -------- ------- ------
+                 ^         ^      ^
+                 |         |      |
+      content is |         |      |
+      unsigned __|         |      |
+                           |      |
+                 claim register  from operands array at index 0
+    v}
+
+    Immediate instructions operands are constructed in the same
+    way:
+
+    {[
+      let im = signed imm op.(1)
+    ]}
+
+    Also one may create variables for convenience:
+
+    {[
+      let x = unsigned var halfword
+    ]}
+
+    that is a creation of variable of bitwidth 16. Other useful
+    bitwidthes are bit, byte, word, doubleword, quadroword. And
+    also it's possible to create a variable of arbitrary bitwidth:
+
+    {[
+      let x = signed var (bitwidth 10)
+    ]}
+
+    It's also possible to create an epxression from integer constant:
+
+    {[
+      let x = unsigned const word 42
+    ]}
+
+    And create an expression from string:
+
+    {[
+      let x = unsigned of_string "0xFFFF_FFFF_FFFF"
+    ]}
+
+    {4 Taking a part}
```

Comment:
"Taking a part" is a colloquial expression that means "to participate in something", it is better to say, to extract a portion of a value. 

Also, make sure that you do not confuse two terms - expression and value. It is common when you see an expression as a value, and people do make such assumptions. But when you say "to extract a portion of expression" (or to take its part, whatever), I think of a poor expression that is taken apart, like you're extracting a variable, or a plus sign from an expression.